autotest/gdrivers/heif.py

```diff
@@ -117,6 +117,26 @@ def test_heif_rgba_16bit():
     assert ds.GetRasterBand(1).DataType == gdal.GDT_UInt16
 
 
+def _has_tiling_support():
+    drv = gdal.GetDriverByName("HEIF")
+    return drv and drv.GetMetadataItem("HEIF_SUPPORTS_TILES")
+
+
+def test_heif_tiled():
+    if not _has_tiling_support():
+        pytest.skip()
+
+    ds = gdal.Open("data/heif/uncompressed_comp_RGB_tiled.heif")
+    assert ds
+    assert ds.RasterXSize == 30
+    assert ds.RasterYSize == 20
+    assert ds.RasterCount == 3
+    assert ds.GetRasterBand(1).DataType == gdal.GDT_Byte
+    assert ds.GetRasterBand(1).GetBlockSize() == [15, 5]
+    assert ds.GetRasterBand(2).GetBlockSize() == [15, 5]
+    assert ds.GetRasterBand(3).GetBlockSize() == [15, 5]
```

Comment:
it could be worth to actually read pixel values, to make sure IReadBlock() is correctly implemented. I understand that libheif might not always return exact pixel values but we could use statistics-based testing like done above in test_heif_exif_endian()

Reply:
This should be occurring through the existing tests (since they all go through the same API, if its available).

I guess there is a corner case where it works for the single-tile case but is off for the M by N case though.

Reply:
> I guess there is a corner case where it works for the single-tile case but is off for the M by N case though.

yes, that was my thinking that you could possibly have got wrong taking into account non-zero nBlockXOff/nBlockYOff. Perhaps not now, but the next guy in 5 years that will touch that code ;-)